include/battle.h

```diff
@@ -456,6 +456,7 @@ struct MegaEvolutionData
     bool8 playerSelect;
     u8 triggerSpriteId;
     bool8 isWishMegaEvo;
+    bool32 didMegaEvo[PARTY_SIZE][2]; // For each party member and side.
```

Comment:
Why a bool32? That looks wasteful. Plus, there should be an easier way to keep track if a mon is a Mega. I'm thinking a species flag should suffice.
Another thought that I've had is that now that we have form change handling, Mega Evolution and Primal Reversion handling could be refactored to work like them, saving resources in BattleStruct.

Reply:
> Why a bool32? That looks wasteful.

I thought it would let the GBA's CPU process it faster because it's better suited to handle that data type :eyes:

> Plus, there should be an easier way to keep track if a mon is a Mega. I'm thinking a species flag should suffice.

Sounds like a better idea, as soon as #2293 is merged, I'll update this PR accordingly :eyes:

> Another thought that I've had is that now that we have form change handling, Mega Evolution and Primal Reversion handling could be refactored to work like them, saving resources in BattleStruct.

I thought that was eventually going to happen ever since we talked about eventually moving all the form changes there in the Discord server. I think it's a bit above what I'm personally capable of though since the Mega Evolution code touches some scary parts of the battle engine's code, so if you wanted to handle that... :eyes:

Reply:
> I thought it would let the GBA's CPU process it faster because it's better suited to handle that data type eyes
While true, it's not a big enough of an improvement

![image](https://user-images.githubusercontent.com/2904965/192600490-0c168ffb-220b-46d8-ba58-b53667bf0144.png)

> Sounds like a better idea, as soon as #2293 is merged, I'll update this PR accordingly eyes

:+1: 

> I thought that was eventually going to happen ever since we talked about eventually moving all the form changes there in the Discord server. I think it's a bit above what I'm personally capable of though since the Mega Evolution code touches some scary parts of the battle engine's code, so if you wanted to handle that... eyes

Yes, that change would be out of the scope of this PR, so this will be handled later.

I'll mark this PR as a draft for now until #2293 is merged